game/scoring.cpp

```diff
@@ -0,0 +1,129 @@
+#include "yahtzee.h"
+#include <algorithm>
+// Util functions
+int sum(int die[5]) {
+    int sum = 0;
+    for (int i = 0; i < 5; ++i)
+    {
+        sum += die[i];
+    }
+    return sum;
+}
+
+//Funtions for scoring yahtzee hand
+
+// Out is an array initialized to zero in which to hold the count
+int sortDie(int die[5], int out[6]) {
```

Comment:
These are `C` style arrays, prefer `C++` style`: `std::vector<int>`. You can pass the value in by pointer as well, and then you won't need a return value.